tonic/src/codec/compression.rs

```diff
@@ -104,19 +126,34 @@ impl CompressionEncoding {
         let header_value = map.get(ACCEPT_ENCODING_HEADER)?;
         let header_value_str = header_value.to_str().ok()?;
 
-        split_by_comma(header_value_str).find_map(|value| match value {
-            #[cfg(feature = "gzip")]
-            "gzip" => Some(CompressionEncoding::Gzip),
-            #[cfg(feature = "zstd")]
-            "zstd" => Some(CompressionEncoding::Zstd),
-            _ => None,
-        })
+        // Get the highest priority supported encoding
+        split_by_comma(header_value_str)
+            // We allow for +1 to account for the identity encoding
+            .take(COMPRESSION_ENCODINGS_LENGTH + 1)
+            .filter_map(|value| {
+                let encoding = match value {
+                    #[cfg(feature = "gzip")]
+                    "gzip" => Some(CompressionEncoding::Gzip),
+                    #[cfg(feature = "zstd")]
+                    "zstd" => Some(CompressionEncoding::Zstd),
+                    _ => None,
+                };
+                if let Some(encoding) = encoding {
+                    enabled_encodings
+                        .priority(encoding)
+                        .map(|priority| (encoding, priority))
+                } else {
+                    None
+                }
+            })
+            .max_by_key(|(_, priority)| *priority)
```

Comment:
I'm confused by this. It seems to suggest that we pick the max by priority (which is implemented as index in the array), but in the other comment you said that the array represents highest-to-lowest priority order. So which is it?